src/api/procedures/createNftCollection.ts

```diff
@@ -396,23 +409,26 @@ async function prepareStorageForCustomType(
     customTypeData = {
       rawId,
       rawValue,
+      isAlreadyCreated: true,
     };
   } else if (!values<string>(KnownNftType).includes(nftType)) {
     const rawValue = stringToBytes(nftType, context);
     const rawId = await context.polymeshApi.query.asset.customTypesInverse(rawValue);
     if (rawId.isNone) {
-      throw new PolymeshError({
-        code: ErrorCode.DataUnavailable,
-        message:
-          'createNftCollection procedure was given a custom type string that does not have a corresponding ID. Register the type and try again',
-        data: { nftType },
-      });
+      const rawCustomAssetTypeId = await context.polymeshApi.query.asset.customTypeIdSequence();
+      const nextCustomAssetTypeId = u32ToBigNumber(rawCustomAssetTypeId).plus(1);
+      customTypeData = {
+        rawId: bigNumberToU32(nextCustomAssetTypeId, context),
+        rawValue,
+        isAlreadyCreated: false,
+      };
```

Comment:
Like we do for Asset ID (where we calculate the next Asset ID to be generated and then batch the other txs with this ID), I have calculated the next custom asset type ID to be generated and later used the same while creating Asset (After registering the custom asset ID first)